ot3_state_manager/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# interface library for headers generated from python for messages
+# use with target_link_libraries(YOURTARGET ot3_state_manager)
+
+set(GENERATED_MESSAGE_HEADER ${CMAKE_BINARY_DIR}/common/simulation/ot3_state_manager_messages.hpp)
+set(PYSRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ot3_state_manager)
+add_custom_command(
+	OUTPUT ${GENERATED_MESSAGE_HEADER}
+	COMMENT "Generating header file and storing to ${GENERATED_MESSAGE_HEADER}"
+	VERBATIM
+	DEPENDS ${PYSRC_DIR}/messages.py
+	COMMAND ${Poetry_EXECUTABLE} run python3 generate_state_manager_headers.py ${GENERATED_MESSAGE_HEADER}
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+	)
+add_custom_target(state-manager-headers DEPENDS ${GENERATED_MESSAGE_HEADER})
+# By depending on this target, you'll be able to include the generated file
+# with no path qualifiers (i.e. #include "ot3_state_manager_messages.hpp")
+add_library(ot3_state_manager INTERFACE)
+add_dependencies(ot3_state_manager state-manager-headers)
+target_include_directories(ot3_state_manager INTERFACE ${CMAKE_CURRENT_BINARY_DIR})
+
+add_custom_target(
+	ot3-state-manager-setup
```

Comment:
if you make this run `COMMAND ${Poetry_EXECUTABLE} config virtualenvs.path = ${CMAKE_BINARY_DIR}/virtualenvs --local` then we'll get the virtualenv in the binaries dir, and we won't need to teardown (you can remove that dir as you always do if you want)

Reply:
675b0c4ad94de42806749397190d29a69a3d2d29